packages/@lwc/wire-service/src/property-trap.ts

```diff
@@ -109,17 +111,18 @@ export function getReactiveParameterValue(
 /**
  * Installs setter override to trap changes to a property, triggering the config listeners.
  * @param cmp The component
- * @param reactiveParameter Reactive parameter that defines the property to monitor
+ * @param reactiveParameters Reactive parameters with the same head, that defines the property to monitor
  * @param configContext The service context
  */
 export function installTrap(
     cmp: EventTarget,
-    reactiveParameter: ReactiveParameter,
+    reactiveParameters: Set<ReactiveParameter>,
     configContext: ConfigContext
 ) {
-    const callback = updated.bind(undefined, cmp, reactiveParameter, configContext);
-    const newDescriptor = getOverrideDescriptor(cmp, reactiveParameter.head, callback);
-    Object.defineProperty(cmp, reactiveParameter.head, newDescriptor);
+    const callback = updated.bind(undefined, cmp, reactiveParameters, configContext);
+    const reactiveParameterHead = reactiveParameters.values().next().value.head;
```

Comment:
Pass the `head` as an argument to this function, it will make it more type-safe, there is no guaranty that the `reactiveParameters` set is non-empty at this point.

Reply:
yep, i like it more that way. thanks!